### PR TITLE
Group By Using Dot Notation

### PR DESCRIPTION
**Pull request by hassanzohdy**
Hello,

This PR allow users when using `SelectInput` or `CheckPicker` to use **groupBy** prop using `dot notation` 

Example Of Usage:

```tsx
const options = [
  {
    label: 'Cairo',
    value: '1',
    country: {
      id: '1',
      name: 'Egypt'
    }
  },
  {
    label: 'Alexandria',
    value: '2',
    country: {
      id: '1',
      name: 'Egypt'
    }
  },
  {
    label: 'New York',
    value: '3',
    country: {
      id: '2',
      name: 'USA'
    }
  },
  {
    label: 'Washington',
    value: '4',
    country: {
      id: '2',
      name: 'USA'
    }
  }
];

export default function HomePage() {
  return (
    <>
      <SelectPicker placeholder="City" groupBy="country.name" data={options} />
    </>
  );
}
```

### CODE REVIEW

File: src/utils/getDataGroupBy.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { Obj } from '@mongez/reinforcements';
```
Comment (SevenOutman) [inline src/utils/getDataGroupBy.ts:1]: You may not have to involve this package as se already have `lodash` as a dependency and you can use its `get` method.

**Review by SevenOutman — CHANGES_REQUESTED** [review]
Hi @hassanzohdy, thanks for this edit! Here's something I'd like to suggest you change, regarding our [contribution guidelines](https://github.com/rsuite/rsuite/blob/main/CONTRIBUTING.md).

- The `package-lock.json` should not be upgraded to `lockfileVersion: 2`
- Changes in codebase should be covered with tests

I will review this PR again after these changes are applied

File: src/utils/getDataGroupBy.ts (hunk 1 above)
Comment (hassanzohdy) [reply src/utils/getDataGroupBy.ts]: @SevenOutman 

I removed the package and updated the package-lock.json file

Have a check on it now.

Regards,

**Comment by hassanzohdy** [comment]
@SevenOutman

I removed the package and updated the package-lock.json file

Have a check on it now.

Regards,

**Review by SevenOutman — COMMENTED** [review]
Could you add test cases for this `getDataGroupBy` function to cover dot-notation use cases?

**Comment by hassanzohdy** [comment]
I Updated `utils/test/getDataGroupBySpec.js` file with the test case for the dot notation syntax.

**Comment by SevenOutman** [comment]
> I Updated `utils/test/getDataGroupBySpec.js` file with the test case for the dot notation syntax.

The test failed ([see logs](https://github.com/rsuite/rsuite/runs/5547134555?check_suite_focus=true#step:7:2740)). Please check whether the test case or implementation is incorrect.

**Comment by hassanzohdy** [comment]
I've updated the test results now it should be working just fine.

If you could tell me how i would run the test as when i run `npm run test` it just go through all tests without any results output.

**Comment by SevenOutman** [comment]
The tests are passing. Congrats! However, the overall coverage has decreased due to the line addition in `src/MaskedInput/createTextMaskInputElement.ts`, could you make sure this change is necessary?

> If you could tell me how i would run the test as when i run `npm run test` it just go through all tests

Setting environment variable `F` to path of a test file make it possible to only run that specific file. And in addition to `npm run test` which runs tests once, `npm run tdd` runs the tests and watches changes in test file and source file and run the tests again automatically.

    F=src/utils/test/getDataGroupBySpec.js npm run tdd

> without any results output.

It's required to have Chrome installed to be able to run the tests locally, but I'm not sure it's relavant.

**Comment by hassanzohdy** [comment]
Thanks for your feedback,

I just reverted it, it was just  typescript compile hint error, i reverted it back to the original

**Review by SevenOutman — APPROVED** [review]
LGTM now :)

**Pull request merged**